it/controllers/pensions/annualAllowance/PensionProviderTaxPaidAnnualAllowanceControllerISpec.scala

```diff
@@ -0,0 +1,488 @@
+    val noEntryAmountErrorMessage = "Enter the amount paid, or that needs to be paid, in pounds"
+  }
+
+  object ExpectedIndividualCY extends SpecificExpectedResults {
+    val expectedTitle = "Did your pension schemes pay or agree to pay the tax?"
+    val expectedHeading = "Did your pension schemes pay or agree to pay the tax?"
+    val expectedErrorTitle = s"Error: $expectedTitle"
+    val amountAboveLimitErrorMessage = "The amount of tax your pension provider paid or agreed to pay must be less than £100,000,000,000"
+    val validAmountErrorMessage = "Enter the amount paid, or that needs to be paid, in the correct format"
+    val noEntryAmountErrorMessage = "Enter the amount paid, or that needs to be paid, in pounds"
+  }
+
+  object ExpectedAgentEN extends SpecificExpectedResults {
+    val expectedTitle = "Did your client’s pension schemes pay or agree to pay the tax?"
+    val expectedHeading = "Did your client’s pension schemes pay or agree to pay the tax?"
+    val expectedErrorTitle = s"Error: $expectedTitle"
+    val amountAboveLimitErrorMessage = "The amount of tax your client’s pension provider paid or agreed to pay must be less than £100,000,000,000"
+    val validAmountErrorMessage = "Enter the amount paid, or that needs to be paid, in the correct format"
+    val noEntryAmountErrorMessage = "Enter the amount paid, or that needs to be paid, in pounds"
+  }
+
+  object ExpectedAgentCY extends SpecificExpectedResults {
+    val expectedTitle = "Did your client’s pension schemes pay or agree to pay the tax?"
+    val expectedHeading = "Did your client’s pension schemes pay or agree to pay the tax?"
+    val expectedErrorTitle = s"Error: $expectedTitle"
+    val amountAboveLimitErrorMessage = "The amount of tax your client’s pension provider paid or agreed to pay must be less than £100,000,000,000"
+    val validAmountErrorMessage = "Enter the amount paid, or that needs to be paid, in the correct format"
+    val noEntryAmountErrorMessage = "Enter the amount paid, or that needs to be paid, in pounds"
+  }
+
+  val userScenarios: Seq[UserScenario[CommonExpectedResults, SpecificExpectedResults]] = Seq(
+    UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedIndividualEN)),
+    UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN, Some(ExpectedAgentEN)),
+    UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY, Some(ExpectedIndividualCY)),
+    UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY, Some(ExpectedAgentCY))
+  )
+
+  private implicit val url: Int => String = pensionProviderTaxPaidAnnualAllowanceUrl
+
+  ".show" should {
+    userScenarios.foreach { user =>
+      import Selectors._
+      import user.commonExpectedResults._
+
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+
+        "render did your pension schemes pay or agree to pay the tax with no pre-filling" which {
```

Comment:
Suggest rename to "render page ... with no pre-filling", as we already know which page is concerned.

Reply:
Renamed the test case name

Reply:
Confirmed.